Support Swift 3 and Xcode 8.

yoichitgy opened the pull request:
Migrated the project to support Swift 3 and Xcode 8.

This pull request consists of 
- https://github.com/yoichitgy/SwinjectPropertyLoader/pull/1 by @gasthana0
- Some cherry-picked commits of #3

yoichitgy committed:
Fix code signing issue on a framework built with Xcode 8.
https://github.com/Swinject/Swinject/issues/120

yoichitgy committed:
Update .travis.yml to use xcode8 image.

yoichitgy committed:
Remove the test in Debug configuration from .travis.yml to reduce CI time.

Commit:
Migrating to swift3, also fixes unit tests.

yoichitgy committed:
Merge pull request #1 from gasthana0/master

Migrating to swift3, also fixes unit tests.

yoichitgy committed:
Specify SWIFT_VERSION at the project level.

yoichitgy committed:
Update jspahrsummers/xcconfigs by Carthage.

yoichitgy commented:
Need to fix .travis.yml.

yoichitgy committed:
Fix .travis.yml. Simulator for iOS 8 is commented out for a simulator issue not running tests.

yoichitgy commented on Sources/ResolverType+Properties.swift:
```diff
@@ -9,26 +9,26 @@
         get {
-            guard let obj = self as? AnyObject else {
-                fatalError("Property feature is not supported unless self is AnyObject.")
-            }
```
Need to check this part can be removed completely.

jakubvano replied:
If this constrain would be required, it is cleaner to express it as
```
extension ResolverType where Self: AnyObject {
    ...
}
```

yoichitgy commented on Sources/ResolverType+Properties.swift:
```diff
@@ -9,26 +9,26 @@
-    private func setProperties(newProperties: [String: AnyObject]) {
-        guard let obj = self as? AnyObject else {
-            fatalError("Property feature is not supported unless self is AnyObject.")
-        }
```
Need to check this part can be removed completely.

yoichitgy commented:
Thanks @jakubvano, I guess I added the `AnyObject` check by my misunderstanding to use an objc associated object. I'll check.

yoichitgy committed:
Remove commented out code for unnecessary type check.
The type check was misunderstanding about AnyObject, NSObject and requirement of objc_get/setAssociatedObject.

yoichitgy commented:
Removed the commented out code by be48d02 because the type check was just misunderstanding about AnyObject, NSObject and requirement of objc_get/setAssociatedObject.
